README.md

```diff
@@ -569,6 +569,8 @@ Note: This example is equivalent to not specifying a mapping or specifying an em
 
 When a `ClusterWorkloadResourceMapping` is defined in the cluster matching a workload resource it **MUST** be used to map the binding that type. If no mapping is available for the type, the implementation **MUST** treat the workload resource as a `PodSpec`-able type.
 
+If a `ClusterWorkloadResourceMapping` resource is modified, and workload resources have had binding data injected using that mapping, then any future modifications to these workload resources **MUST** use the new `ClusterWorkloadResourceMapping` definitions.  Existing workloads **MUST NOT** be modified.
```

Comment:
this is the opposite of my recollection from the last community call. Mandating that resources under management never be modified in perpetuity is asking for trouble.

IIRC the discussion was: during the next reconciliation, existing bindings are removed using the previous mapping and then reapplied using the new mapping. How the previous mapping is tracked through a change is left as an implementation detail (stashing as an annotation on the workload resource is likely the simplest solution).

Reply:
I was trying to specify that an operator shouldn't reproject all the service bindings for that workload type automatically when a new mapping is defined.  IIRC, this was a pitfall we were trying to avoid, and this was my attempt on making that behavior clear.  Perhaps I could have made that clearer in my wording.

One thing I'm not entirely clear on: what mechanism are we going to use to have users redo their existing bindings?  Should we let them update/patch existing bindings?  Or should we force them to remove and recreate their bindings?  I'm not inclined to either solution here; could we allow both?

Reply:
I think it would be good to state the user scenario where someone would update the workload mapping.  The two cases that jump to mind are:
1. The referenced workload `kind` now has a new version, so the workload mapping CR is updated to have the v.next mappings
2. The user realizes they made a mistake when defining the mapping, and update the CR to correct it

Scenario 1 is simple **if** the workload mapping start with "v1", then adds "v2", etc, so then there's no effect on existing workload projections.  It gets tricky if the mapping started with a version wildcard "*" and then was updated to have specific mappings for versions - which then I think it should affect existing workload projections.

Scenario 2 should always clear out the old projections (which were incorrect) and perform the new projections.

So it may not be too bad to specify at the spec level what happens. At the implementation level though, it probably does imply the controller has to keep watching any workload mapping CRs that were used, so that it can update the projections if user scenario 1 or 2 happens.